Kinvey.Core/Utils/HelperMethods.cs

```diff
@@ -73,6 +73,49 @@ public static string getRequestID(IRestResponse response)
 			return string.Empty;
 		}
 
+        public static string GetRequestStartTime(IRestResponse response)
+        {
+            string XKinveyRequestStart = string.Empty;
+
+            if (response != null && response.Headers != null)
+            {
+                try
+                {
+                    var item = response.Headers
+                                       .Cast<Parameter>()
+                                       .SingleOrDefault(i => i.Name.ToLower().Equals(Constants.STR_HEADER_REQUEST_START_TIME))
+                                       .Value;
+
+                    Type valueType = item.GetType();
+                    if (valueType != null)
+                    {
+                        if (valueType.GetElementType() == typeof(string))
+                        {
+                            XKinveyRequestStart = item.ToString();
+                        }
+                        else if (valueType.IsConstructedGenericType)
```

Comment:
@vinaygahlawat i don't get it. why do we need to check the type here? the header value should not always be a `string`?

Reply:
It's a good question. At times during debugging, some response headers have come back as other types, such as `List<string>`, and so this code is defensive, to make sure we are a bit more robust about how we cast. But as for the reason why it is different at times, I'm not sure yet.